### PR TITLE
[PTQ] batch_size in PTQ docs

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Add documentation regarding batch_size

### Reason for changes

To meet users with influence of batch_size on quantization

### Related tickets

121650

### Tests

N/A

### CODE REVIEW

File: docs/compression_algorithms/post_training/Quantization.md [hunk 1]
```diff
@@ -91,3 +91,22 @@ NNCF provides the examples of Post-Training Quantization where you can find the
...
+When utilizing post-training quantization with NNCF, the recommendation is to use a ```batch_size``` of 1 for your dataloader.
+
+However, NNCF supports dataloaders with arbitrary batch sizes, but there are some limitations that need to be taken into consideration. For some architectures, such as transformer-like models, statistics collection for ```batch_size``` greater than 1 could lead to inaccurate results.
+
+If the results after quantization with a batch size greater than 1 are inaccurate, the following recommendations apply:
+
+1) Set the ```batch_size``` to 1 for the dataloader.
+2) In ```AdvancedQuantizationParameters```, set the ```batchwise_statistics``` parameter to ```False```.
+
+Why are the statistics results different? During statistics collection for quantization with ```batch_size``` greater than 1 in NNCF, the assumption of having the batch axis lying on the 0-axis is made. This is not a general case, but for many models, it is applicable. For models where this assumption does not fit, it leads to inaccurate statistics.
```
Comment (MaximProshin) [inline docs/compression_algorithms/post_training/Quantization.md:108]: I'm not sure if it's really needed in the documentation, but I leave for others review and up to you. The rest LGTM.

**Review by MaximProshin — APPROVED** [review]

File: docs/compression_algorithms/post_training/Quantization.md [hunk 2]
```diff
@@ -91,3 +91,22 @@ NNCF provides the examples of Post-Training Quantization where you can find the
 function: [PyTorch](../../../examples/post_training_quantization/torch/mobilenet_v2/README.md), [TensorFlow](../../../examples/post_training_quantization/tensorflow/mobilenet_v2/README.md), [ONNX](../../../examples/post_training_quantization/onnx/mobilenet_v2/README.md), and [OpenVINO](../../../examples/post_training_quantization/openvino/mobilenet_v2/README.md)
 
 In case the Post-Training Quantization algorithm could not reach quality requirements you can fine-tune a quantized pytorch model. Example of the Quantization-Aware training pipeline for a pytorch model could be found [here](../../../examples/quantization_aware_training/torch/resnet18/README.md).
+
+## Dataloader Usage
```
Comment (alexsu52) [inline docs/compression_algorithms/post_training/Quantization.md:95]:
```suggestion
## Using `pytorch.Dataloader` or `tf.data.Dataset` as data source for calibration dataset.
```
I think you can add a general information about what kind `dataloader` are supported and how NNCF handle batch size. 

If you want to give only information about only batch size support then you can call topic as following: "Batch size support in calibration dataset."

File: docs/compression_algorithms/post_training/Quantization.md [hunk 3]
```diff
@@ -91,3 +91,22 @@ NNCF provides the examples of Post-Training Quantization where you can find the
...
+
+```batch_size``` is a parameter of a dataloader that refers to the number of samples or data points propagated through the neural network in a single pass.
+
+When utilizing post-training quantization with NNCF, the recommendation is to use a ```batch_size``` of 1 for your dataloader.
```
Comment (alexsu52) [inline docs/compression_algorithms/post_training/Quantization.md:99]: It depends on the model.

File: docs/compression_algorithms/post_training/Quantization.md [hunk 4]
```diff
@@ -91,3 +91,22 @@ NNCF provides the examples of Post-Training Quantization where you can find the
...
+
+However, NNCF supports dataloaders with arbitrary batch sizes, but there are some limitations that need to be taken into consideration. For some architectures, such as transformer-like models, statistics collection for ```batch_size``` greater than 1 could lead to inaccurate results.
+
+If the results after quantization with a batch size greater than 1 are inaccurate, the following recommendations apply:
```
Comment (alexsu52) [inline docs/compression_algorithms/post_training/Quantization.md:103]: What kind of model does NNCF support quantization with calibration dataset with batch_size > 1.

File: docs/compression_algorithms/post_training/Quantization.md [hunk 5]
```diff
@@ -91,3 +91,22 @@ NNCF provides the examples of Post-Training Quantization where you can find the
...
+
+1) Set the ```batch_size``` to 1 for the dataloader.
+2) In ```AdvancedQuantizationParameters```, set the ```batchwise_statistics``` parameter to ```False```.
```
Comment (alexsu52) [inline docs/compression_algorithms/post_training/Quantization.md:106]: What is `batchwise_statistics`?

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: docs/compression_algorithms/post_training/Quantization.md (hunk 2 above)
Comment (kshpv) [reply docs/compression_algorithms/post_training/Quantization.md]: Applied

File: docs/compression_algorithms/post_training/Quantization.md (hunk 4 above)
Comment (kshpv) [reply docs/compression_algorithms/post_training/Quantization.md]: Add description of unsupported models

File: docs/compression_algorithms/post_training/Quantization.md (hunk 5 above)
Comment (kshpv) [reply docs/compression_algorithms/post_training/Quantization.md]: I am not sure whether we need to expose this based on new changes (add a clear explanation of unsupported models)

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**